net/relayd/src/opnsense/service/conf/actions.d/actions_relayd.conf

```diff
@@ -0,0 +1,29 @@
+[start]
+command:/usr/local/etc/rc.d/relayd start
+type:script
+message:starting relayd
+
+[stop]
+command:/usr/local/etc/rc.d/relayd stop
+type:script
+message:stopping relayd
+
+[status]
+command:/usr/local/etc/rc.d/relayd status || exit 0
```

Comment:
"... status; exit 0" preferred

Reply:
Always __exit 0__ ? Even on errors?

Reply:
exit 0 is used here if status fails which it can when the service is not running. it's a bit of a silly interface of the rc.d script. in any case "|| exit 0" and "; exit 0" do the same thing so making it more explicit  use "; exit 0"

Reply:
Silly? No it's common usage of exit codes.
Most programs exit with 0 if everything is OK. So the rc.d scripts.
But OPNsense interprets a 0 as error.
A "; exit 0" will always return 0. The "status" would always fail.
"|| exit 0" returns 0 only if the rc.d script returns a code that is not 0.
The "status" would always fail. too. :confused:

Reply:
After reading _/usr/local/opnsense/service/modules/processhandler.py_ I'm even more confused:

> 464:                    if exit_status == 0:
> 465:                        return 'OK'
> 466:                    else:
> 467:                        syslog.syslog(syslog.LOG_ERR, '[%s] returned exit status %d' %
 (message_uuid, exit_status))
> 468:                        return 'Error (%d)' % exit_status

So an "; exit 0" let the status always succeed.

Reply:
nope:

exit 1 || exit 0 = exit 0
exit 1; exit 0 = exit 0

"||" is used programmatically, here we just fudge the status command to behave as exit 1 would cause configd to think the command failed when it really doesn't.... it returns text we evaluate either way.

Reply:
Oh, I've completely misunderstood.
With the status command we are not interested in the exit code and to avoid confusing configd we return always 0. Then "; exit 0" makes sense.

Reply:
configd dispatches commands and logs everything that fails... so far so good

the "status" command in rc.d scripts fails when the service is not running, but returns 0 when the service is running so that creates an issue for configd even though it just wanted to return the status (it's not a failure that a service is not running from our point of view because we don't know and that's why we query).

Reply:
yep :)